lib/routes/smh/author.ts

```diff
@@ -0,0 +1,66 @@
+import { Route } from '@/types';
+import ofetch from '@/utils/ofetch';
+import { load } from 'cheerio';
+
+export const route: Route = {
+    path: '/by/:authorSlug',
+    categories: ['programming'],
+    example: '/smh/by/natassia-chrysanthos-h17jwj',
+    parameters: { authorSlug: 'Slug of the author profile URL' },
+    name: 'Articles by an author',
+    maintainers: ['yue-dongchen'],
+    handler: async (ctx) => {
+        const baseUrl = 'https://smh.com.au';
+        const { authorSlug = 'natassia-chrysanthos-h17jwj' } = ctx.req.param();
```

Comment:
Since https://github.com/DIYgod/RSSHub/blob/e82cbba47904631d9738ed8a3a8f26347e977745/lib/routes/smh/author.ts#L6 defines `authorSlug` as a compulsory parameter, no default value should be provided.